chain-extensions/pallet-assets/src/tests.rs

```diff
@@ -0,0 +1,459 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::mock::*;
+use frame_support::assert_ok;
+use frame_support::traits::Currency;
+use pallet_contracts::{CollectEvents, DebugInfo, Determinism};
+use pallet_contracts_primitives::{Code, ExecReturnValue};
+use parity_scale_codec::Encode;
+use sp_core::crypto::AccountId32;
+use sp_runtime::DispatchError;
+use std::fs;
+
+// Those tests use the assets chain extension example available here:
+// https://github.com/AstarNetwork/ink-test-contracts
+// It maps chain extension functions to ink! callable messages
+// ex:
+// #[ink(message)]
+// pub fn burn(&mut self, asset_id: u128, who: AccountId, amount: Balance) -> Result<(), AssetsError> {
+//    AssetsExtension::burn(asset_id, who, amount)?;
+//     Ok(())
+// }
+
+#[test]
+fn mint_works() {
+    ExtBuilder::default()
+        .existential_deposit(50)
+        .build()
+        .execute_with(|| {
+            let addr = instantiate();
+
+            // Arrange - create asset and give contract mint permission (Issuer role)
+            assert_ok!(Assets::create(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                ALICE,
+                1
+            ));
+            assert_ok!(Assets::set_team(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                addr.clone(),
+                ALICE,
+                ALICE
+            ));
```

Comment:
Out of curiosity - is this needed for the test?

Reply:
Yes it is needed. Only EOA can create assets (because create is not part of CE) and set_team is used to transfer Admin roles. As in Chain extension the Origin can only be the contract, Issuer (minter role) should be transferred to it in order to mint assets.
I move the comment line on top of it for more clarity

Reply:
One thing you could do in test is use `force_create`, that way you can set the owner to whoever in a single call.

This is more information, no changes needed.